yaml.go

```diff
@@ -196,6 +211,15 @@ func convertToJSONableObject(yamlObj interface{}, jsonTarget *reflect.Value) (in
 		}
 	}
 
+	// Transform map[string]interface{} into map[interface{}]interface{}
```

Comment:
Is the need for this new? What is the cost of this?

Reply:
Yes, yamlv3 unmarshals maps into `map[string]interface{}` if all keys are strings, while yamlv2 did unmarshal all maps into `map[interface{}]interface{}`.

Reply:
JSON only deals with map[string]interface{}, and it will marshal the same… do we need to convert back to map[interface{}]interface{}?